dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESIndexAPI.java

```diff
@@ -134,38 +133,58 @@ public String getStatus() {
 	}
 
 	public ESIndexAPI(){
-		this.esclient = new ESClient();
 		this.iapi = new ContentletIndexAPIImpl();
-		this.esIndexHelper = ESIndexHelper.INSTANCE;
-		this.serverAPI = APILocator.getServerAPI();
+		this.esIndexHelper = ESIndexHelper.getInstance();
 		this.clusterAPI = APILocator.getClusterAPI();
 	}
 
-	@VisibleForTesting
-	protected ESIndexAPI(final ESClient esclient, final ContentletIndexAPIImpl iapi, final ESIndexHelper esIndexHelper,
-						 final ServerAPI serverAPI, final ClusterAPI clusterAPI){
-		this.esclient = esclient;
-		this.iapi = iapi;
-		this.esIndexHelper = esIndexHelper;
-		this.serverAPI = serverAPI;
-		this.clusterAPI = clusterAPI;
-	}
+    private class IndexSortByDate implements Comparator<String> {
+        public int compare(String o1, String o2) {
+            if (o1 == null || o2 == null) {
+                return 0;
+            }
+            if (o1.indexOf("_") < 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [String.indexOf(char) is faster than String.indexOf(String).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)